finat/point_set.py

```diff
@@ -72,6 +72,9 @@ def indices(self):
 
     @cached_property
     def expression(self):
+        npoints, _ = self.points.shape
+        if npoints <= 1:
+            return gem.Literal(self.points[0])
```

Comment:
Now this one special cases `PointSet` to behave like `PointSingleton` in case of the single point.  I kind of suspect this is going to cause trouble because the index that is there in the general case, will now be missing in a particular case...

Reply:
> Now this one special cases `PointSet` to behave like `PointSingleton`

like the original `PointSingleton`, before the changes that are being proposed here.

Reply:
Good point